src/test/java/seedu/address/logic/parser/AddressBookParserTest.java

```diff
@@ -73,10 +78,22 @@ public void parseCommand_exit() throws Exception {
 
     @Test
     public void parseCommand_find() throws Exception {
-        List<String> keywords = Arrays.asList("foo", "bar", "baz");
-        FindCommand command = (FindCommand) parser.parseCommand(
-                FindCommand.COMMAND_WORD + " " + keywords.stream().collect(Collectors.joining(" ")));
-        assertEquals(new FindCommand(new NameContainsKeywordsPredicate(keywords)), command);
+        List<String> nameList = List.of("John", "Alex");
+        List<String> personIdList = List.of("B13", "C28");
+        FindGuestCommand command = (FindGuestCommand) parser.parseCommand(
+                FindGuestCommand.COMMAND_WORD + " "
+                    + nameList.stream().map(p -> PREFIX_NAME + p)
+                        .collect(Collectors.joining(" ")) + " "
+                    + personIdList.stream().map(p -> PREFIX_ID + p)
+                        .collect(Collectors.joining(" "))
+        );
+        Predicate<Person> expectedPredicate = new MatchNamePredicate(nameList
+            .stream().map(p -> new Name(p)).collect(Collectors.toList()))
+            .or(new MatchPersonIdPredicate(personIdList
+                .stream().map(p -> new PersonId(p)).collect(Collectors.toList()))
+            );
+        //now that command is a composite predicate, can't test like this.
+        //assertEquals(new FindGuestCommand(expectedPredicate), command);
```

Comment:
I would suggest letting parser parse the command and put some hand-written names and put it to predicates of both sides instead of using assertEquals directly.

Reply:
Just like you said, the or in predicate make comparison seem impossible, so should we drop the test entirely?

Reply:
Oh you can write some manual tests for both predicates (parsed one, constructed one), instead of using "assertEquals" directly
.

Reply:
I have change the structure of findGuestCommand